tests/s3_storage_handler_test.py

```diff
@@ -0,0 +1,782 @@
+        - The Moto server and logger setup are common to both test cases.
+    """
+    server = ThreadedMotoServer()
+    server.start()
+    secrets = {"s3endpoint": endpoint, "accesskey": "", "secretkey": "", "region": "sbg"}
+
+    logger = logging.getLogger("s3_storage_handler_test")
+    logger.setLevel(logging.DEBUG)
+    logger.handlers = []
+    logger.addHandler(logging.StreamHandler(sys.stdout))
+
+    server.stop()
+    if endpoint == "http://localhost:5000":
+        assert S3StorageHandler(secrets["accesskey"], secrets["secretkey"], secrets["s3endpoint"], secrets["region"])
+    else:
+        with pytest.raises(Exception):
+            assert S3StorageHandler(
+                secrets["accesskey"],
+                secrets["secretkey"],
+                secrets["s3endpoint"],
+                secrets["region"],
+            )
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "s3cfg_file, expected_res",
+    [(("./USER/credentials_location/.s3cfg", True)), (("/path/to/none/.s3cfg", False))],
+)
+def test_get_secrets(s3cfg_file: str, expected_res: bool):
+    """Docstring to be added."""
+    secrets = {
+        "s3endpoint": None,
+        "accesskey": None,
+        "secretkey": None,
+    }
+    logger = logging.getLogger("s3_storage_handler_test")
+    logger.setLevel(logging.DEBUG)
+    logger.handlers = []
+    logger.addHandler(logging.StreamHandler(sys.stdout))
+
+    if "USER" in s3cfg_file:
+        tmp_s3cfg_file, tmp_path = tempfile.mkstemp()
+        try:
+            with os.fdopen(tmp_s3cfg_file, "w") as tmp:
+                tmp.write("access_key = test_access_key\n")
+                tmp.write("secret_key = test_secret_key\n")
+                tmp.write("host_bucket = https://test_endpoint.com\n")
+                tmp.flush()
+            ret = S3StorageHandler.get_secrets(secrets, tmp_path, logger)
+            assert expected_res == ret
+        except OSError:
+            assert False
+        finally:
+            os.remove(tmp_path)
+    else:
+        assert expected_res == S3StorageHandler.get_secrets(secrets, s3cfg_file, logger)
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "path, expected_res",
+    [(("/usr/path/to/file", "file")), (("/usr/path/to/folder/", "folder"))],
+)
+def test_get_basename(path: str, expected_res: bool):
+    """test_get_basename Function Documentation
+
+    Test the get_basename method of the S3StorageHandler class.
+
+    Parameters:
+    - path (str): The input path to be tested.
+    - expected_res (bool): The expected result of the get_basename method.
+
+    Raises:
+    - AssertionError: If the result of S3StorageHandler.get_basename(path) does not match expected_res.
+    """
+    assert expected_res == S3StorageHandler.get_basename(path)
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "endpoint, bucket, nb_of_files",
+    [
+        (("http://localhost:5000", "test-bucket", 2012)),
+        (("http://localhost:5000", "test-bucket", 999)),
```

Comment:
Could we use less files so the test takes less time ? 21 and 9 files would be sufficient instead of 2012 and 999 ?

Reply:
No, this is exactly the point of this test. The paginator can handle a max of 1000 files